File: validation/validation-annotation-processor/src/test/java/ru/tinkoff/kora/validation/annotation/processor/testdata/Foo.java
package ru.tinkoff.kora.validation.annotation.processor.testdata;

import ru.tinkoff.kora.validation.common.annotation.NotEmpty;
import ru.tinkoff.kora.validation.common.annotation.Pattern;
import ru.tinkoff.kora.validation.common.annotation.Range;
import ru.tinkoff.kora.validation.common.annotation.Validated;

import javax.annotation.Nullable;
import java.time.OffsetDateTime;

@Validated
public record Foo(@NotEmpty @Pattern("\\d+") String number,
                  @Range(from = 1, to = 10, boundary = Range.Boundary.INCLUSIVE_EXCLUSIVE) long code,
                  @Nullable OffsetDateTime timestamp,
                  @Validated @Nullable Bar bar) {}

File: validation/validation-symbol-processor/src/test/kotlin/ru/tinkoff/kora/validation/symbol/processor/testdata/Foo.kt
package ru.tinkoff.kora.validation.symbol.processor.testdata

import ru.tinkoff.kora.validation.common.annotation.NotEmpty
import ru.tinkoff.kora.validation.common.annotation.Pattern
import ru.tinkoff.kora.validation.common.annotation.Range
import ru.tinkoff.kora.validation.common.annotation.Validated
import java.time.OffsetDateTime

@Validated
data class Foo(
    @NotEmpty @Pattern("\\d+")
    val number: String,
    @Range(from = 1.0, to = 10.0)
    val code: Long,
    val timestamp: OffsetDateTime,
    @Validated
    val bar: Bar?
)
